first commit

jacobperia opened the pull request.

jacobperia committed:
first commit

danielpaul commented on app/controllers/jobs_controller.rb:
```diff
@@ -0,0 +1,58 @@
+class JobsController < ApplicationController
+  before_action :get_company
```
```suggestion
  before_action :set_company
```

danielpaul commented on app/controllers/jobs_controller.rb:
```diff
@@ -0,0 +1,58 @@
+
+  def destroy
+    authorize @job
+    @job.destroy
```
if destroy, redirect. Otherwise, redirect back to the index page saying that it could not be deleted and to try again. 

There could be validations that we could add that would prevent a record to be deleted similar to other validations. (We don't have this now, but we could later and our app needs to be able to support that).

danielpaul commented on app/models/job.rb:
```diff
@@ -0,0 +1,9 @@
+
+  has_rich_text :description
+
+  enum job_type: { 'Full-time' => 0, 'Part-time' => 1 }
```
the hash-rockets format (`=>`) is an older format. Let's use the newer version of `:`. Also this would be the best way to do enums:

`enum localization: { home: 0, foreign: 1, none: 2 }`

Where the names are keys and not strings.

danielpaul commented on app/views/companies/show.html.haml:
```diff
@@ -1,14 +1,43 @@
+          -if policy(@company).update?
+            %td= link_to 'Edit', edit_company_job_path(@company, job)
+            %td= link_to 'Delete', company_job_path(@company, job), method: :delete, data: { confirm: 'Are you sure?' }
+%br
```
better space the code. you can leave empty lines in the code to separate sections of the code better

danielpaul commented on bin/yarn:
```diff
@@ -3,7 +3,7 @@ APP_ROOT = File.expand_path('..', __dir__)
   yarn = ENV["PATH"].split(File::PATH_SEPARATOR).
     select { |dir| File.expand_path(dir) != __dir__ }.
-    product(["yarn", "yarn.exe"]).
+    product(["yarn", "yarn.cmd", "yarn.ps1"]).
```
what was this for? was it because of any issues you had?

jacobperia replied:
It got modified when i installed a new gem, probably faker or image processing?

danielpaul commented on config/routes.rb:
```diff
@@ -1,5 +1,8 @@
   devise_for :users, controllers: { registrations: 'registrations' }
-  resources :companies
+  resources :companies do
+    get '/companies/:id', to: 'jobs#index'
```
```suggestion
```

A better way for this would be that we'd use the companies controller for the show method. 

And then the jobs#index you can redirect from the controller to the company#show. 

It would be not intuitive to overwrite resource routes to another controller.

danielpaul commented on spec/factories/user.rb:
```diff
@@ -1,10 +1,21 @@
     confirmed_at { DateTime.now }
   end
 end
+# FactoryBot.define do
```
let's delete code that is not needed. We can always get back the code if we ever need it later via Git histroy.

danielpaul commented on spec/models/job_spec.rb:
```diff
@@ -0,0 +1,25 @@
+  describe 'company creation specs' do
+    it 'should be able to save job only with a user who the ocmpany belongs to' do
+      user = build(:user)
+      user.save!
```
you can avoid `.save` commands here by calling `create` instead of `build` in the line above

danielpaul commented on spec/models/job_spec.rb:
```diff
@@ -0,0 +1,25 @@
+      expect(job.save).to eq(true)
+    end
+
+    it 'should not be able to save company without a user' do
```
is this means to be without a valid company?

danielpaul commented:
okey cool.

On Sun, Apr 18, 2021 at 23:39:19, jacobperia < ***@***.*** > wrote:

> 
> 
> 
> ***@***.**** commented on this pull request.
> 
> 
> 
> In bin/ yarn (
> https://github.com/IdentitySquare/jobs-board/pull/3#discussion_r615464433
> ) :
> 
> > @@ -3,7 +3,7 @@ APP_ROOT = File.expand_path('..', __dir__)
> 
> Dir.chdir(APP_ROOT) do
>    yarn = ENV["PATH"].split(File::PATH_SEPARATOR).
> 
> select { |dir| File.expand_path(dir) != __dir__ }.
> -    product(["yarn",
> "yarn.exe"]).
> +    product(["yarn", "yarn.cmd", "yarn.ps1"]).
> 
> 
> It got modified when i installed a new gem, probably faker or image
> processing?
> 
> 
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub (
> https://github.com/IdentitySquare/jobs-board/pull/3#discussion_r615464433
> ) , or unsubscribe (
> https://github.com/notifications/unsubscribe-auth/AACRLMNRHYTU43CITRZW25DTJNNRPANCNFSM43DLCBCA
> ).
> 
> 
>

jacobperia committed:
resolved PR issues
create instead of build
fixed routes
fixed readability
added few new tests
other small method name changes

jacobperia committed:
forgot and end line in test

danielpaul commented on app/models/job.rb:
```diff
@@ -0,0 +1,9 @@
+
+  has_rich_text :description
+
+  enum job_type: { 'Full-time': 0, 'Part-time': 1 }
```
We won't be using strings for the keys of enums. 

```suggestion
  enum job_type: { full_time: 0, part_time: 1 }
```

jacobperia committed:
fixed enum in model, fixed views to display enum properly
